tensorflow/core/kernels/mkl_conv_ops.cc

```diff
@@ -603,98 +864,72 @@ class MklConv2DOp : public OpKernel {
                                           memory::format::hwio);
       filter.SetUsrMem(filter_md, &filter_tensor);
 
-      // Set output shape (output_dims) required in MKL-DNN order.
-      // Currently, we set output layout as Tensorflow's layout (NHWC or NCHW
-      // depending on data format). But later we propagate Mkl layout of the
-      // output to the next op directly.
-      output.SetUsrMem(output_dims_mkl_order, tf_fmt);
-
-      // Create memory descriptors for convolution data w/ no specified format.
-      src.SetOpMemDesc(src_dims, memory::format::any);
-      filter.SetOpMemDesc(filter_dims, memory::format::any);
-      output.SetOpMemDesc(output_dims_mkl_order, memory::format::any);
-
       // MKLDNN dilation starts from 0.
       dilations[kDilationH] -= 1;
       dilations[kDilationW] -= 1;
 
+      // get a conv2d fwd from primitive pool
+      Conv2DFwd<T> *conv2d_fwd = nullptr;
+      memory::dims bias_dims = {};
+      if (biasEnabled) {
+        conv_utl.GetBiasSizeInMklOrder(kInputIndex_Bias, &bias_dims);
+        conv2d_fwd = Conv2DFwdFactory<T>::Get(src_dims, filter_dims,
+                        bias_dims, dst_dims_mkl_order,
+                        strides, dilations, padding_l, padding_r);
+      } else {
+        conv_utl.GetBiasSizeInMklOrder(kInputIndex_Bias, &bias_dims);
```

Comment:
It looks like this line is not needed.

Reply:
Cut/paste error. Fixed by deleting the line.